### PR TITLE
All output channels must be set to null before shutdown

### PR DESCRIPTION
**Pull request by computerquip-streamlabs**

### CODE REVIEW

File: app/services/sources/sources.ts [hunk 1]
```diff
@@ -170,6 +170,17 @@ export class SourcesService extends StatefulService<ISourcesState> implements IS
...
+     * sure we reset the channel it's set to,
+     * otherwise OBS thinks it's still attached
+     * and won't release it. */
+    const channel = this.state.sources[id].channel;
+
+    /* Be consistent with initial check */
+    if (channel !== void 0) {
+      obs.Global.setOutputSource(channel, null);
+    }
+
```
Comment (holiber) [inline app/services/sources/sources.ts]: Just to simplify put the code above `const source = this.getSource(id);`
```
const source = this.getSource(id);
 if (source.channel !== void 0) {
    obs.Global.setOutputSource(channel, null);
 }
```

Comment (computerquip-streamlabs) [reply app/services/sources/sources.ts]: Fixed in forced push.

File: app/services/scenes-transitions.ts [hunk 2]
```diff
@@ -64,6 +64,11 @@ export class ScenesTransitionsService extends StatefulService<ISceneTransitionsS
     this.getCurrentTransition().release();
   }
 
+  reset() {
+    this.getCurrentTransition().release();
```
Comment (andycreeth) [inline app/services/scenes-transitions.ts]: NBD, but for DRY-ness this could just be `this.release()`

File: app/services/sources/sources.ts [hunk 3]
```diff
@@ -171,6 +171,15 @@ export class SourcesService extends StatefulService<ISourcesState> implements IS
 
   removeSource(id: string) {
     const source = this.getSource(id);
+
+   /* When we release sources, we need to make
```
Comment (andycreeth) [inline app/services/sources/sources.ts]: This is misaliagned.

**Pull request merged**